nsv13/code/modules/overmap/fighters/_fighters.dm

```diff
@@ -1588,6 +1618,53 @@ Utility modules can be either one of these types, just ensure you set its slot t
 	burst_size = 3
 	fire_delay = 0.5 SECONDS
 
+/obj/item/fighter_component/primary/laser
+	name = "Stinger Class Phaser Cannon"
+	icon_state = "lasercannon"
+	weight = 3 //it's a laser. it's light, but the gun is fuckhueg
+	accepted_ammo = null
+	overmap_select_sound = 'nsv13/sound/effects/ship/phaser_adjust.ogg'
+	overmap_firing_sounds = list('nsv13/sound/effects/ship/burst_phaser.ogg', 'nsv13/sound/effects/ship/burst_phaser2.ogg')
+	burst_size = 3
+	fire_delay = 10 SECONDS
+	var/projectile = /obj/item/projectile/beam/laser/phaser
+	var/charge_to_fire = 2000   // this is probably not final. needs to be a good balance between having an actual mainship weapon on a fighter, and firing one shot every minute because one shot depletes your entire battery
+
+/obj/item/fighter_component/primary/laser/get_ammo()
+	var/obj/structure/overmap/small_craft/F = loc
+	if(!istype(F))
+		return FALSE
+	var/obj/item/fighter_component/battery/B = F.loadout.get_slot(HARDPOINT_SLOT_BATTERY)
+	if(!istype(B))
+		return 0
+	return B.charge
+
+/obj/item/fighter_component/primary/laser/get_max_ammo()
+	var/obj/structure/overmap/small_craft/F = loc
+	if(!istype(F))
+		return FALSE
+	var/obj/item/fighter_component/battery/B = F.loadout.get_slot(HARDPOINT_SLOT_BATTERY)
+	if(!istype(B))
+		return 0
+	return B.maxcharge
+
+/obj/item/fighter_component/primary/laser/fire(obj/structure/overmap/target)
+	var/obj/structure/overmap/small_craft/F = loc
+	if(!istype(F))
+		return FALSE
+	var/obj/item/fighter_component/battery/B = F.loadout.get_slot(HARDPOINT_SLOT_BATTERY)
+
+	if(B.charge < charge_to_fire)
+		F.relay('sound/weapons/gun_dry_fire.ogg')
+		return FALSE
+
+	var/datum/ship_weapon/SW = F.weapon_types[fire_mode]
+	SW.default_projectile_type = projectile
+	SW.fire_fx_only(target, lateral = TRUE)
+	B.charge -= charge_to_fire
+	return TRUE
+
+
 /obj/item/fighter_component/secondary
 	name = "Fuck you"
```

Comment:
Funny. No. I challenge you to find any other location in this codebase where the word "fuck" is somewhere player facing. Fix this.

Reply:
I wasn't really going to mess with basetypes of anything